osu.Game.Rulesets.Catch/Skinning/LegacyComboCounter.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Screens.Play;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+using static osu.Game.Skinning.LegacySkinConfiguration;
+
+namespace osu.Game.Rulesets.Catch.Skinning
+{
+    /// <summary>
+    /// A combo counter implementation that visually behaves almost similar to osu!stable's combo counter.
+    /// </summary>
+    public class LegacyComboCounter : CompositeDrawable, ICatchComboCounter
+    {
+        private readonly ISkin skin;
+
+        private readonly string fontName;
+        private readonly float fontOverlap;
+
+        private readonly LegacyRollingCounter counter;
+
+        public LegacyComboCounter(ISkin skin)
+        {
+            this.skin = skin;
+
+            fontName = skin.GetConfig<LegacySetting, string>(LegacySetting.ComboPrefix)?.Value ?? "score";
+            fontOverlap = skin.GetConfig<LegacySetting, float>(LegacySetting.ComboOverlap)?.Value ?? -2f;
+
+            AutoSizeAxes = Axes.Both;
+
+            Alpha = 0f;
+            Anchor = Anchor.Centre;
+            Origin = Anchor.Centre;
+            Scale = new Vector2(0.8f);
+
+            InternalChild = counter = new LegacyRollingCounter(skin, fontName, fontOverlap)
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+            };
+        }
+
+        private IBindable<bool> isBreakTime;
+
+        [Resolved(canBeNull: true)]
+        private GameplayBeatmap beatmap { get; set; }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            isBreakTime = beatmap?.IsBreakTime.GetBoundCopy();
+            isBreakTime?.BindValueChanged(b =>
```

Comment:
is this honestly required? seems like we could just drop this code and reduce the complexity of this PR by miles. the only difference is the combo counter will stay on screen for up to 1 second extra during the start of a break?

Reply:
Yeah thought about that initially, but didn't want to change how it behaves on osu!stable, and didn't think it would make the PR that complex (but turns out it did), will remove it when I have time to.

Reply:
it can of course be added back once we know what we're doing with break time knowledge.